src/CodeGen_Hexagon.cpp

```diff
@@ -2324,11 +2183,9 @@ bool CodeGen_Hexagon::possiblyCodeGenWideningMultiply(const Mul *op) {
     Ops.clear();
     //   Vdd.uh=vmpy(Vu.ub,Rt.ub)
     //   Vdd.uw=vmpy(Vu.uh,Rt.uh)
-    Patterns.push_back(Pattern(WPICK(wild_u16x128, wild_u16x64) *
-                               WPICK(wild_u16x128, wild_u16x64),
+    Patterns.push_back(Pattern(wild_u16x2W * wild_u16x2W,
```

Comment:
That is interesting, the element type also seems to be the widened output element type (as opposed to the input element type)...probably need to have Pranav comment on this one since this is code he wrote.  Also looks like the same situation exists with the signed patterns at lines 2149-2152 also in this routine.

As an experiment, I tried changing this to single width "W" and the vmpy codegen test Halide/test/hexagon/codegen/vmpy.cpp starts failing so I think I would leave the patterns as is for now since they are producing an expected result. 

```
/local/mnt/scratch/dpalermo/tot-work/halide/Halide-0309/test/hexagon/codegen/vmpy.cpp:22:11: error: expected string not found in input
 //CHECK: vmpy(v{{[0-9]+}}.ub,r{{[0-9]+}}.b)
          ^
<stdin>:21290:2: note: scanning from here
 vmem(r12+#0) = v0.new
 ^
<stdin>:21326:11: note: possible intended match here
 v3:2.w = vmpy(v2.h,v3.h)
```